docs/csharp/tutorials/exploration/records.md

```diff
@@ -0,0 +1,154 @@
+
+## Compute degree days
+
+To compute degree days, you take the difference from a baseline temperature and the mean temperature on a given day. To measure heat over time, you discard any days where the mean temperature is below the baseline. To measure cold over time, you discard any days where the mean temperature is above the baseline. For example, the U.S. uses 65F as the base for both heating  and cooling degree days. That's the temperature where no heating or cooling is needed. If a day has a mean temperature of 70F, that day is 5 cooling degree days and 0 heating degree days. Conversely, if the mean temperature is 55F, that day is 10 heating degree days and 0 cooling degree days.
+
+You can express these formulas as a small hierarchy of record types: an abstract degree day type and two concrete types for heating degree days and cooling degree days. These types can also be positional records. They'll take a baseline temperature and a sequence of daily temperature records as arguments to the primary constructor:
+
+:::code language="csharp" source="snippets/record-types/InterimSteps.cs" ID="DegreeDaysRecords":::
+
+The abstract `DegreeDays` record is the shared base class for both the `HeatingDegreeDays` and `CoolingDegreeDays` records. The primary constructor declarations on the derived records show how to manage base record initialization. Your derived record declares parameters for all the parameters in the base record primary constructor. The base record declares and initializes those properties. The derived record doesn't hide them, but only creates and initializes properties for arguments that aren't declared in its base record. In this example, the derived records don't add new primary constructor arguments. Test your code by adding the following code to your `Main` method:
+
+:::code language="csharp" source="snippets/record-types/Program.cs" ID="HeatingAndCooling":::
+
+You'll get output like the following display:
+
+```dotnetcli
+HeatingDegreeDays { BaseTemperature = 65, TempRecords = record_types.DailyTemperature[], DegreeDays = 85 }
+CoolingDegreeDays { BaseTemperature = 65, TempRecords = record_types.DailyTemperature[], DegreeDays = 71.5 }
+```
+
+Your code calculates the correct number of heating and cooling degree days over that period of time. But this example shows why you may want to replace some of the synthesized methods for records. The `TempRecords` element in the display isn't useful. It displays the type, but nothing else. You can change this behavior by providing your own implementation of the synthesized `PrintMembers` method. The compiler generates this method for all record types, unless you write your own. The signature depends on modifiers applied to the `record` declaration:
+
+- If the `record` type is `sealed`, the signature is `private sealed bool PrintMembers(StringBuilder builder);`
```

Comment:
In most of the article, records are referred to without backticks:
```suggestion
- If a record type is `sealed`, the signature is `private bool PrintMembers(StringBuilder builder);`
```

Also no `sealed` modifier on the method:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEDMACKcIBs4AmGCcAxgPYJECCA3ALABQQA===